apm-agent-core/src/test/java/co/elastic/apm/agent/impl/circuitbreaker/CircuitBreakerTest.java

```diff
@@ -190,10 +190,13 @@ void testPauseThroughConfigThenResumeOnlyWhenStressRelieved() throws IOException
 
         // 3 stress should keep it paused
         monitor.simulateStress();
+        // needs to poll all monitors otherwise tracer state might not be up-to-date within tracer (only updated once polling is over)
+        assertAllMonitorsPolled(()->{}, monitor);
         assertSteadyState(this::assertPaused, monitor);
```

Comment:
This is the change what fixes the test flakyness, as it will now
- wait for all stress monitors to be polled, which will then update tracer state
- wait for all stress monitors to be polled at the beginning of steady paused state.

As tracer stress status isn't atomic, there is a race condition between "all stress monitors has been polled" and "update tracer stress state".